.pre-commit-config.yaml

```diff
@@ -17,6 +17,14 @@ repos:
         name: pytest
         entry: poetry run bash scripts/test-coverage-xml.sh
         language: system
-        files: ^src/zenml/|^tests/
+        files: ^src/zenml/|^tests/|^examples/
         types: [file, python]
-        pass_filenames: false
\ No newline at end of file
+        pass_filenames: false
+
+  -   repo: https://github.com/codespell-project/codespell
+      rev: v2.1.0
+      hooks:
+      -   id: codespell
+          name: codespell
+          description: Checks for common misspellings in text files.
+          entry: codespell -c -I .codespell-ignore-words -f -i 0 --builtin clear,rare,en-GB_to_en-US,names,code CODE-OF-CONDUCT.md CONTRIBUTING.md ROADMAP.md README.md RELEASE_NOTES.md src/ docs/book examples/
```

Comment:
Two small improvements here:
- the spell checking is now run twice if we modify some python file right? Once inside the previous linting step and here? Can we maybe remove it from the linting script in this case?
- Pre-commit automatically passes all the files modified in this commit to the script that gets run (unless you pass `pass_filenames: false`. So if we change the entry to `codespell -c -I .codespell-ignore-words -f -i 0 --builtin clear,rare,en-GB_to_en-US,names,code` it should automatically only run on the files that were modified inside the commit

Reply:
@schustmi When you say "remove it from the linting script", do you mean remove codespell completely from the linting script?